pynamodb/attribute_dict.py

```diff
@@ -0,0 +1,33 @@
+import collections
+
+
+class AttributeDict(collections.MutableMapping):
+    """
+    A dictionary that stores attributes by two keys
+    """
+    def __init__(self, *args, **kwargs):
+        self._values = {}
+        self._alt_values = {}
```

Comment:
this has to do with the fact that pynamo lets you pass in an `attr_name` ... the name in the db might be different than the name in the code (e.g. couponCode in db, coupon_code in py code) 

garrett also suggested that i figure out a way to mask this complexity from other parts of the code. i think if i do that though it will incur an O(n) search through the keys when i need to look for the one that i don't index off of, but i may be wrong